src/main/java/frc/robot/util/SparkAdapter.java

```diff
@@ -104,29 +108,42 @@ public SparkBaseConfig newConfig() {
     }
 
     @Override
-    public SparkAdapter newAdapter(int deviceID) {
-      return new SparkAdapter(new SparkFlex(deviceID, spark.getMotorType()));
+    public SparkAdapter newAdapter(String namePrefix, int deviceID) {
+      return new SparkAdapter(namePrefix, new SparkFlex(deviceID, spark.getMotorType()));
     }
   }
 
+  private static final DataLog LOG = DataLogManager.getLog();
+
+  private final String namePrefix;
   private final Accessor spark;
 
+  private final DoubleLogEntry logOutputCurrent;
+  private final DoubleLogEntry logTemperature;
+
   /**
    * Constructs a SparkAdapter for a {@link SparkMax} motor controller.
    *
    * <p>This private constructor is intended for use only by delegating public constructors or
    * factory methods. It assumes the SparkMax object is already configured or will be configured
    * approriately by the public constructors or factory methods.
    *
+   * @param namePrefix The prefix for the log entries.
    * @param spark The SparkMax object to adapt.
    */
-  private SparkAdapter(SparkMax spark) {
+  private SparkAdapter(String namePrefix, SparkMax spark) {
+    this.namePrefix = namePrefix;
     this.spark = new SparkMaxAccessor(spark);
+
+    String name = String.format("%s/SparkMax-%d", namePrefix, spark.getDeviceId());
+    this.logOutputCurrent = new DoubleLogEntry(LOG, "%s/OutputCurrent", name);
+    this.logTemperature = new DoubleLogEntry(LOG, "%s/Temperature", name);
```

Comment:
Did you really intend for name to be set as metadata on these logs?
```suggestion
    this.logOutputCurrent = new DoubleLogEntry(LOG, name + "/OutputCurrent");
    this.logTemperature = new DoubleLogEntry(LOG, name + "/Temperature");
```

Reply:
Nope. Good catch!